blog/content/post/2020-09-08-raspberrypi_docker.md

```diff
@@ -0,0 +1,295 @@
+Here it comes for the rescue Raspberry Pi working with Docker! With these tools,
+you can create a lot of projects with small effort and low cost.
+
+### What is Raspberry Pi?
+
+Raspberry Pi is a representation of a small computer. It has all the basic
+hardware components like memory, processor, etc. and extensions like USB, Wi-FI,
+sound controller. Components of this device depending on the model that
+you are using. Because of littleness (size similar to credit-card) and cheapness,
+it is a very popular device to do everything that you would expect from
+a typical PC. For example, you can play videos, browse the internet,
+or even play some simple games. However, don't expect the Raspberry Pi
+to be as good as a PC. It would have some issues loading complex websites
+or running more aggravating games. The possibilities of this device are
+infinitely great. For example with Raspberry Pi you can create:
+
+- Retro Gaming Console,
+- Google Home,
+- The Parent Detector,
+- Weather Station,
+- Robots
+
+and a lot more!
+
+### What is Docker?
+
+Docker open source project helping developers to create, deploy,
+and run applications a lot easier by concept named 'containers.'
+They are lightweight execution environments sharing operating system kernel.
+Except for this case they are running in isolation.
+Containers allow us to package up libraries, dependencies, etc. in one place.
+By that, we can assume that the application will run on any other device
+without worries of possible customized settings of this particular machine.
+
+___
+
+### How to run Docker on Raspberry Pi?
+
+Installation Docker on Raspberry Pi is really fast and easy. Follow these steps:
+
+Download:
+```bash
+curl -fsSL https://get.docker.com -o get-docker.sh
+```
+
+Execute script:
+```bash
+sh get-docker.sh
+```
+
+This script will detect OS, install packages, and run Docker.
+This operation may take a while so you need to wait. In the end,
+the script will output all necessary information about Docker and how to use it.
+Additionally, if you want to run commands without prepending sudo
+(as a non-root user) you have to add the user name to the Docker group.
+
+Add user to Docker group:
+```bash
+sudo usermod -aG docker \$USER
+```
+
+**Note:** ```$USER``` variable holds your username.
+It comes from the environment.
+
+Now Docker is set up on Raspberry Pi and finally,
+you can do some cool stuff!
+
+### How to use Docker?
+
+The basic concepts of Docker are Images and Containers. The first of them
+is made of filesystem layers that make a possibility to execute applications.
+Inside it are source code, dependencies, libraries, tools, or instructions for
+installing and running an application. Dockerfile image is just a binary file
+that is immutable for the user. The second one - containers are mutable beings.
+They are using Docker Images like templates. If you are a developer you can
+think that Docker Image is a class and Container is an instance
```

Comment:
Images and containers may be compared to classes and instances in object-oriented programming. [nie wiem czy to ma sens]